keps/sig-auth/3766-referencegrant/README.md

```diff
@@ -160,89 +164,13 @@ object).
-they are providing access to. If that authorization check fails, the update or
-create action will also fail.
-
-<<[UNRESOLVED Do we need checks beyond read access? ]>>
-Previous Discussion: https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084528657
-
-Comment that started that thread:
-
-does anything ensure the user creating the ReferenceGrant has permissions (read?
-write?) on the object they are granting access to? Translating the existing
-ReferenceGrant into an authz check means translating from Kind to Resource
-
-since this is extending "half of a handshake", it seems important to ensure the
-actor extending the handshake actually has permission to do so
-
-<<[/UNRESOLVED]>>
-
-#### `Resource` vs `Kind`
-
-When creating a metaresource (that is, a resource that targets other resources)
-like ReferenceGrant, it's important to consider if the metaresource uses the more
-common `Kind` or the more correct `Resource`.
-
-In the original Gateway API implementation, we chose to use `Kind` rather than
-`Resource`, mainly to improve the user experience. That is, it's easier users
-to take the value from the `kind` field at the top of the YAML they are already
-using, and put it straight into these fields, rather than needing to do a
-kind-resource lookup for every user's interaction with the API. @robscott even
-ended up making https://github.com/kubernetes/community/pull/5973 to clarify
-the API conventions.
-
-However, in discussion on this KEP, it's clear that the more generic nature of
-_this_ API requires the additional specificity that `Resource` provides.
-
-The Gateway API ReferenceGrant looked like this:
-```yaml
-apiVersion: gateway.networking.k8s.io/v1beta1
-kind: ReferenceGrant
-metadata:
-  name: allow-gateways
-  namespace: bar
-spec:
-  from:
-    # Note that in Gateway API, Group is currently defaulted
-    # to this, which means you to explicitly set the group to
-    # the empty string for Core resources. We should definitely
-    # change this.
-    - group: "gateway.networking.kubernetes.io"
-      kind: Gateway
-      namespace: foo
-  to:
-   - group: ""
-     kind: Secret
-```
-
-The new version will look like this instead:
-```yaml
-apiVersion: authorization.k8s.io/v1alpha1
-kind: ReferenceGrant
-metadata:
-  name: allow-gateways
-  namespace: bar
-spec:
-  from:
-    # Assuming that we leave the default for Group to the empty
-    # string, so that Core objects don't need additional config.
-    - group: "gateway.networking.kubernetes.io" 
-      resource: gateways
-      namespace: foo
-  to:
-    - group: ""
-      resource: secrets
-
-```
-
-The new version communicates the scope more clearly because `group`+`resource`
-is unambiguous and corresponds to exactly one set of objects on the API Server.
+Anyone creating or updating a ReferenceGrant MUST have read access to the
+resources they are providing access to. If that authorization check fails, the
```

Comment:
Is the authorization check going to be enforced by the api server?

Reply:
@robscott 
https://docs.google.com/document/d/1Xa0aNpHIv6udg1IEFiVKIsYZ1WYC4JqoMpmg8XaCRSU/edit#heading=h.hb6enzp5cixq
I referred to the above document.

> Is the authorization check going to be enforced by the api server?
I understood that the authentication check is performed by the API server, is my understanding correct?